app/src/main/java/tech/relaycorp/gateway/data/DataModule.kt

```diff
@@ -74,15 +75,19 @@ class DataModule {
 
     // PoWeb
 
+    @Provides
+    fun dohClient() =
+        // TODO: Remove custom DNS resolver once we can use CloudFlare's (which is the default)
+        // https://github.com/cloudflare/cloudflare-docs/issues/565
+        DoHClient("https://dns.google/dns-query")
```

Comment:
Since I'm not super familiar with Dagger and Android, I should probably highlight what I intend to do here in case I'm missing something:

I want to keep one DoH client around, so that the same instance can be reused when making queries. In the current version of the code, only the PoWeb client and the URL ping use this DoH client. This is to be able to reuse the underlying resources (e.g., connection pool, keep-alive TCP connections) and [eventually caching](https://github.com/relaycorp/doh-jvm/issues/8).

Reply:
To keep it around, just add a dagger Scope to it, and it will be shared inside the scope. If it's the `@Singleton` scope, it will be shared across the whole app.

Reply:
Thanks! I think `@Singleton` will do, having read the docs. I've just added it.